test/unit/job_invocation_test.rb

```diff
@@ -0,0 +1,33 @@
+require 'test_plugin_helper'
+
+describe JobInvocation do
+
+  let(:job_invocation) { FactoryGirl.build(:job_invocation) }
+  let(:template) { FactoryGirl.create(:job_template, :with_input) }
+
+  context 'Able to be created' do
+    it { assert job_invocation.save }
+  end
+
+  context 'Requires targeting' do
+    before { job_invocation.targeting = nil }
+
+    it { refute_valid job_invocation }
+  end
+
+  context 'has template invocations' do
+    let(:input_value) { FactoryGirl.build(:template_invocation_input_value) }
+
+    before do
+      job_invocation.template_invocations << TemplateInvocation.new(:template => template)
```

Comment:
I attempted this for some time and failed and figured my time was better spent elsewhere.  But if you really want this done in factory girl, i can try some more